lib/cocoapods/installer/pod_source_installer.rb

```diff
@@ -117,13 +117,30 @@ def unlock_files!(file_accessors)
       # @return [void]
       #
       def download_source
+        verify_source_is_secure(root_spec)
         download_result = Downloader.download(download_request, root, :can_cache => can_cache?)
 
         if (@specific_source = download_result.checkout_options) && specific_source != root_spec.source
           sandbox.store_checkout_source(root_spec.name, specific_source)
         end
       end
 
+      # Verify the source of the spec is secure, which is used to
+      # show a warning to the user if that isn't the case
+      # This method doesn't verify all protocols, but currently
+      # only prohibits unencrypted http:// connections
+      #
+      def verify_source_is_secure(root_spec)
+        return if root_spec.source.nil? || root_spec.source[:http].nil?
+        http_source = root_spec.source[:http]
+        # `:http` doesn't mean it's "http", might as well be "https://" here
```

Comment:
I'm not sure this comment makes sense in general - unless it's a dig at the key value, in which case it's not really adding much, so it can probably go :P